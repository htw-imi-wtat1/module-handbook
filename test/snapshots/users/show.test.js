const { app, User, request, removeID } = require('../../commonJest')

describe('User Index', function () {
  const userData = {
    name: { first: 'Jamila', last: 'Marl' },
    email: 'laurenz.schwandke69_@yahoo.com',
    zipCode: 94577,
    password: 'Ffit82AW5EvioKC'
  }
  let userID // this is shared state between all tests!
  beforeAll(function (done) {
    User.create(userData)
      .then(created => {
        userID = created.id
        done()
      })
      .catch(error => {
        // console.log('error caught: ' + error.message)
        done(error.message)
      })
  })
  it('renders user index correctly', (done) => {
    request(app)
      .get(`/users/${userID}`)
      .then((response) => {
        expect(removeID(response.text, userID)).toMatchSnapshot()
        done()
      })
  })
})
