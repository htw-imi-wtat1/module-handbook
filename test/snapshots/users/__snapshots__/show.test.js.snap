// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Index renders user index correctly 1`] = `
"<!DOCTYPE html>
<html>

<head>
	<meta charset=\\"utf-8\\">
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=0.8, maximum-scale=1\\">
	<title>Module Handbook</title>
	<link rel=\\"shortcut icon\\" type=\\"image/png\\" href=\\"/images/icon.png\\" />
	<link rel=\\"stylesheet\\" href=\\"/css/bootstrap.css\\">
	<link rel=\\"stylesheet\\" href=\\"/css/module_handbook.css\\">
	<script src=\\"/jquery.min.js\\"></script>
	<script src=\\"/popper.js\\"></script>
	<script src=\\"/js/bootstrap.min.js\\"></script>
</head>

<body>
<div class = \\"container\\">
	<div id=\\"nav\\">
		<div class=\\"col-sm nav-align\\">
			<h1 id=\\"title\\">Module Handbook</h1>
		</div>
		<nav class=\\"navbar navbar-expand-lg navbar-light bg-light\\">
    <a class=\\"navbar-brand\\" href=\\"/\\">Home</a>
    <button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\"#navbarSupportedContent\\"
            aria-controls=\\"navbarSupportedContent\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\">
        <span class=\\"navbar-toggler-icon\\"></span>
    </button>

    <div class=\\"collapse navbar-collapse\\" id=\\"navbarSupportedContent\\">
        <ul class=\\"navbar-nav mr-auto\\">
            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/about\\">About
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/modules/list\\">Module List
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/modules/tabular\\">Module Table
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/courses\\">Courses
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/users\\">Users
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/users/new\\">Register
        
    </a>
</li>

            


                    
                       <li class=\\"nav-item\\">
                          <a  class=\\"nav-link\\" href =\\"/users/login\\">Log In</a>
                       </li>
                    

        </ul>

    </div>
</nav>
		

<div class=\\"modal fade\\" id=\\"cookieBanner\\" tabindex=\\"-1\\" role=\\"dialog\\" aria-labelledby=\\"exampleModalLabel\\" aria-hidden=\\"true\\">
    <div class=\\"modal-dialog\\">
    <div class=\\"modal-content\\">
    <div class=\\"modal-header\\">
    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">Data Privacy Warning and Cookie Consent</h5>
<button type=\\"button\\" class=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\">
    <span aria-hidden=\\"true\\">&times;</span>
</button>
</div>
<div class=\\"modal-body\\">
This is an example website for teaching and learning the MERN stack.
This site uses cookies
to store session state and handle user authentication.
It is not fully functional and may have bugs. Do not enter any
personal or sensitive data on this site!
</div>

<div class=\\"modal-footer\\">
    <button type=\\"button\\" class=\\"btn btn-success\\" data-dismiss=\\"modal\\" onclick=\\"cookiesOK()\\">OK, understood!</button>
    <button type=\\"button\\" class=\\"btn btn-warning\\" data-dismiss=\\"modal\\" onclick=\\"cookiesReadMore()\\">Read More</button>
    <button type=\\"button\\" class=\\"btn btn-danger\\" data-dismiss=\\"modal\\" onclick=\\"cookiesGoodBye()\\">No Thanks. Take me somewhere else.</button>
</div>

</div>
</div>
</div>


<script src=\\"/js/cookieBanner.js\\"></script>
	</div>
	<div class = \\"container\\">
		
		<div class = \\"flashes mt-4\\">
       
		</div>
	
	  <h1 class=\\"center\\">User Data for
  Jamila Marl
</h1>

<table class=\\"table\\">
  <tr>
    <th>Name</th>
    <td>
      Jamila Marl
    </td>
  </tr>
  <tr>
    <th>Email</th>
    <td>
      laurenz.schwandke69_@yahoo.com
    </td>
  </tr>
  <tr>
    <th>Zip Code</th>
    <td>
      94577
    </td>
  </tr>
  <tr>
    <th>Password</th>
    <td>********</td>
  </tr>
</table>


<div class=\\"center\\">
  <a class=\\"button\\" href=\\"/users/<replaced_mongoose_id>/edit\\">
    Edit User Details
  </a> |
  <a class=\\"button\\" href=\\"/users/<replaced_mongoose_id>?_method=DELETE\\" onclick=\\"return confirm('Are you sure you want to delete this record?')\\" a>Delete</a>
 |
  <a class=\\"button\\" href=\\"/users\\">View all users</a>
</div>
<hr/>
<h1>Study LogBook</h1>
<div class=\\"center\\">
    <a class=\\"button\\" href=\\"/users/<replaced_mongoose_id>/log_entries/new\\">
        New Entry
    </a>
</div>
<table class=\\"table table-hover\\">
    <thead>
    <tr>
        <th>Course</th>
        <th>Semester</th>
        <th>Event</th>
        <th>Date</th>
        <th>Notes</th>
        <th colspan=\\"3\\"></th>
    </tr>
    </thead>

    <tbody>
    

    </tbody>
</table>

<br>

	</div>
</div>


</body>

</html>
"
`;
